test/common/config/datasource_test.cc

```diff
@@ -115,7 +116,7 @@ TEST_F(AsyncDataSourceTest, LoadRemoteDataSourceReturnFailure) {
 
   initialize([&](Http::RequestMessagePtr&, Http::AsyncClient::Callbacks& callbacks,
                  const Http::AsyncClient::RequestOptions&) -> Http::AsyncClient::Request* {
-    callbacks.onFailure(Envoy::Http::AsyncClient::FailureReason::Reset);
+    callbacks.onFailure(request_, Envoy::Http::AsyncClient::FailureReason::Reset);
     return nullptr;
```

Comment:
should you be returning the request_ ptr here?

Reply:
Actually, I've decided to restore original `return nullptr`  in all other places.

`return nullptr` affects the code path that follows `AsyncClient::send()`, so I shouldn't have changed it in the first place.